riscv/processor.cc

```diff
@@ -735,11 +752,28 @@ reg_t processor_t::get_csr(int which)
       if (!supports_extension('F'))
         break;
       return state.frm;
-    case CSR_FCSR:
-      require_fp;
-      if (!supports_extension('F'))
-        break;
-      return (state.fflags << FSR_AEXC_SHIFT) | (state.frm << FSR_RD_SHIFT);
+    case CSR_FCSR: {
```

Comment:
According to the current proposal (https://github.com/riscv/riscv-v-spec/pull/135/files), access to FCSR, VXRM, and VXSAT depends only on FS, not VS.  So I think we need to rewrite this block of code as:

      require_fp;
      reg_t val = 0;
      bool is_set = false;

      if (supports_extension('F')) {
        val |= (state.fflags << FSR_AEXC_SHIFT) |
               (state.frm << FSR_RD_SHIFT);

        is_set = true;
      }
      if (supports_extension('V')) {
        val |= (VU.vxrm << FSR_VXRM_SHIFT) |
               (VU.vxsat << FSR_VXSAT_SHIFT);
        is_set = true;
      }

      if (is_set)
        return val;

      break;

Reply:
Similarly, we should change the checks for VXRM and VXSAT to use `require_fp` instead of `require_vector_vs`.

Reply:
Agree with the modification but I will discard this for this PR and update the spike after spec change is merged.
To add "VS" should be important since some developers also question the missing of "VS" for v0.7 spec in Qemu mailing list.